### PR TITLE
(#1206) Updates list reset page to match USWDS

### LINKED ISSUE #1206: Typography subpage: List Reset
Update the List Reset page within the Typography section of the documentation site with the content needed for NCIDS for All pilot launch.

### Sub-Tasks
- [ ] Draft content for page
- [ ] Add images (if needed)
- [ ] Convert to .mdx
- [ ] Update documentation site page

### PR DESCRIPTION
**Pull request by dlescarbeau**
- Closes #1206

### CODE REVIEW

**Review by sarinapadilla — APPROVED** [review]
Looks good to me!

File: docs/content/foundations/typography/list-reset.mdx [hunk 1]
```diff
@@ -10,9 +10,11 @@ utility_modules:
     utility_module_name: 'add-list-reset'
     utility_class_intro: |
       The list reset utility classes have not been modified from the U.S. Web Design System (USWDS). Refer to the [USWDS List reset utility](https://designsystem.digital.gov/utilities/list-reset/#list-reset-2).
-    utility_class_display_component: 'TestUtilityClassTable'
+    utility_class_display_component: 'ListResetDemonstration'
     utility_class_display_params:
-      classes: []
+      classes:
+        - class_name: 'default'
```
Comment (olitharp-nci) [inline docs/content/foundations/typography/list-reset.mdx:16]: `.default` is not a class.

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: docs/src/components/layouts/utility-page-layout/utility-class-tables/list-reset-demonstration.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+/**
+ * This component is for demonstrating the opacity utilities.
+ * @param {Array<OpacityTableDemonstrationParams>} param0.utilityClassDisplayParams
+ * @returns
+ */
+const ListResetDemonstration = ({ utilityClassDisplayParams }) => (
+	<div className="site-utility-examples">
+		<div className="grid-row site-utility-examples__header-row">
```
Comment (olitharp-nci) [inline docs/src/components/layouts/utility-page-layout/utility-class-tables/list-reset-demonstration.jsx:50]: remove uswds-site classes

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: docs/content/foundations/typography/list-reset.mdx (hunk 1 above)
Comment (dlescarbeau) [reply docs/content/foundations/typography/list-reset.mdx:16]: `.default` on USWDS seems to just be a way for them to show the difference between the list reset and a standard list. I should remove this ?

Comment (olitharp-nci) [reply docs/content/foundations/typography/list-reset.mdx:16]: Screenshot 2024-07-29 at 1 48 39 PM

You are telling a user that a class `.default` exists. It does not. Our documentation should probably not be lying to users trying to use it.

Comment (olitharp-nci) [reply docs/content/foundations/typography/list-reset.mdx:16]: Screenshot 2024-07-29 at 1 50 54 PM

USWDS does not have a class here they are just showing it as an example. but notice how there is no classes

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — APPROVED** [review]

**Pull request merged**